hermes-common/src/main/java/pl/allegro/tech/hermes/common/message/wrapper/AvroMessageContentWrapper.java

```diff
@@ -19,24 +21,17 @@
     @SuppressWarnings("unchecked")
     UnwrappedMessageContent unwrapContent(byte[] data, Schema schema) {
         try {
-            GenericRecord record = bytesToRecord(data, schema);
-            Map<Utf8, Utf8> metadata = (Map<Utf8, Utf8>) record.get(METADATA_MARKER);
+            Map<Utf8, Utf8> metadata = (Map<Utf8, Utf8>) bytesToRecord(data, schema).get(METADATA_MARKER);
 
-            long timestamp = Long.parseLong(metadata.remove(METADATA_TIMESTAMP_KEY).toString());
-            String messageId = metadata.remove(METADATA_MESSAGE_ID_KEY).toString();
+            long timestamp = getTimestamp(metadata);
```

Comment:
I think it will be better to check only once if metadata exists instead of checking it in every read field.